url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/StandAloneSimulation.java

```diff
@@ -0,0 +1,527 @@
+			double from = 1;
+			double to = 1;
+			if (paramsU[0].contains("start"))
+			{
+				from = Double.parseDouble(paramsU[0].replace("start", "").replace(":", "").trim());
+				to = Double.parseDouble(paramsU[1].replace("end", "").replace(":", "").trim());
+			}
+			else
+			{
+				from = Double.parseDouble(paramsU[1].replace("start", "").replace(":", "").trim());
+				to = Double.parseDouble(paramsU[0].replace("end", "").replace(":", "").trim());
+			}		
+			Uniform uniform = null;
+			if(caseD.equalsIgnoreCase("start")) {
+				uniform = new Uniform(from, to, true);
+			} else {
+				uniform = new Uniform(from, to, false);
+			}
+			return uniform;
+		case ("Gaussian Mixture Models"):
+			String tempM = prop.getProperty(caseD + "_parameters").replace("[", "").replace("]", "").replace("\"", "");
+			String[] paramsM = tempM.split("}");
+			int length = paramsM.length;//-1;
+			double[] means = new double[length];
+			double[] stds = new double[length];
+			double[] w = new double[length];
+			double sumW = 0;
+			for (int i=0; i<length; i++)
+			{
+				String tempS = paramsM[i].replace("{", "").trim();
+				if (tempS.startsWith(","))
+					tempS = tempS.substring(1);
+				String[] paramsMM = tempS.split(",");
+				if (paramsMM.length != 3 || !(paramsMM[0].contains("mean") || paramsMM[1].contains("mean") || paramsMM[2].contains("mean")) 
+						|| !(paramsMM[0].contains("std") || paramsMM[1].contains("std") || paramsMM[2].contains("std")) 
+						|| !(paramsMM[0].contains("w") || paramsMM[1].contains("w") || paramsMM[2].contains("w")) )
+					throw new Exception("ERROR: Gaussian Mixture Models require exaclty 3 parameters for each tuple, named \"mean\", \"std\" and \"w\". "+
+						" E.g. [{\"w\":0.5 , \"mean\":45,\"std\":10}, {\"w\":0.5 , \"mean\":100,\"std\":10}]");
+				for (int j=0; j<3; j++)
+				{
+					if (paramsMM[j].contains("mean"))
+						means[i] = Double.parseDouble(paramsMM[j].replace("mean", "").replace(":", "").trim());
+					else if (paramsMM[j].contains("std"))
+						stds[i] = Double.parseDouble(paramsMM[j].replace("std", "").replace(":", "").trim());
+					else
+					{
+						w[i] = Double.parseDouble(paramsMM[j].replace("w", "").replace(":", "").trim());
+						sumW += w[i];
+					}
+				}
+			}
+			if (sumW != 1)
+				throw new Exception("ERROR: Gaussian Mixture Models require for tuple weights to sum up to 1");
+			GaussianMixtureModels gmm = new GaussianMixtureModels(length, w, means, stds, true);
+			return gmm;
+		case ("Histogram"):
+			String tempH = prop.getProperty(caseD + "_values").replace("[", "").replace("]", "").replace("\"", "").trim();
+			String[] values = tempH.split(",");
+			if (values.length == 1 && values[0].equals(""))
+				values = new String[0];
+			double[] v = new double[values.length];
+			for (int i=0; i<values.length; i++)
+				try {
+					v[i] = Double.parseDouble(values[i]);
+				} 
+				catch (NumberFormatException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)